packages/common/container-definitions/src/loader.ts

```diff
@@ -102,19 +102,128 @@ export interface ICodeAllowList {
+ * Events emitted by the {@link IContainer} "upwards" to the Loader and Host.
  */
+/* eslint-disable @typescript-eslint/unified-signatures */
 export interface IContainerEvents extends IEvent {
+    /**
+     * Emitted when the readonly state of the container changes.
+     *
+     * @remarks Listener parameters:
+     *
+     * - `readonly`: Whether or not the container is now in a readonly state.
+     *
+     * @see {@link IContainer.readOnlyInfo}
+     */
     (event: "readonly", listener: (readonly: boolean) => void): void;
+
+    /**
+     * Emitted when the {@link IContainer} completes connecting to the Fluid service.
+     *
+     * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+     *
+     * @see
+     *
+     * - {@link IContainer.connectionState}
+     *
+     * - {@link IContainer.connect}
+     */
     (event: "connected", listener: (clientId: string) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `codeDetails`: TODO
+     *
+     * - `proposal`: TODO
+     *
+     * @see {@link IContainer.proposeCodeDetails}
+     */
     (event: "codeDetailsProposed", listener: (codeDetails: IFluidCodeDetails, proposal: ISequencedProposal) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `codeDetails`: TODO
+     */
     (event: "contextChanged", listener: (codeDetails: IFluidCodeDetails) => void);
-    (event: "disconnected" | "attached", listener: () => void);
+
+    /**
+     * Emitted when the {@link IContainer} becomes disconnected from the Fluid service.
+     *
+     * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+     *
+     * @see
+     *
+     * - {@link IContainer.connectionState}
+     *
+     * - {@link IContainer.disconnect}
+     */
+    (event: "disconnected", listener: () => void);
+
+    /**
+     * TODO
+     */
+    (event: "attached", listener: () => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `error`: TODO
+     */
     (event: "closed", listener: (error?: ICriticalContainerError) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `error`: TODO
+     */
     (event: "warning", listener: (error: ContainerWarning) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `message`: TODO
+     */
     (event: "op", listener: (message: ISequencedDocumentMessage) => void);
-    (event: "dirty" | "saved", listener: (dirty: boolean) => void);
+
+    /**
+     * Emitted when the first local change has been made, following a "saved" event.
+     *
+     * @remarks
+     *
+     * The "saved" event will be emitted once all local changes have been acknowledged by the service.
+     *
+     * TODO: what does listener param represent in this case?
```

Comment:
It doesn't appear as though the "dirty" parameter in the callback here is ever used, at least not in our codebase. What is it intended to mean? Is it not redundant with the event invocation itself?

Reply:
Pretty sure it's just legacy.  Not recommended for use.